haddock-library/src/Documentation/Haddock/Types.hs

```diff
@@ -33,7 +33,9 @@ import Data.Bitraversable
 -- meta-data to comments. We make a structure for this ahead of time
 -- so we don't have to gut half the core each time we want to add such
 -- info.
-newtype Meta = Meta { _version :: Maybe Version } deriving (Eq, Show)
+data Meta = Meta { _version :: Maybe Version
+                 , _package :: Maybe Package
+                 } deriving (Eq, Show)
```

Comment:
@alexbiehl I take it this also requires waiting till the next major GHC version, right? Should I move this PR to `ghc-head`?